### PR TITLE
[DataFormats][clang-tidy] make deleted function public

### PR DESCRIPTION
**Pull request by smuzaffar**
We have over 25K clang-tidy warnings about `deleted member function should be public [modernize-use-equals-delete]`. 8K of these come from DataFormats sub-system.
Locally tested, it build and all of the `modernize-use-equals-delete` warnings are fixed.

[a]
```
DataFormats/Common/interface/PtrVectorBase.h:165:20: warning: deleted member function should be public [modernize-use-equals-delete]
    PtrVectorBase& operator=(const PtrVectorBase&) = delete;
                   ^
DataFormats/Common/interface/ContainerMaskTraits.h:40:5: warning: deleted member function should be public [modernize-use-equals-delete]
    ContainerMaskTraits() = delete;
    ^
DataFormats/Common/interface/ContainerMaskTraits.h:41:5: warning: deleted member function should be public [modernize-use-equals-delete]
    ContainerMaskTraits(const ContainerMaskTraits&) = delete;  // stop default
    ^
DataFormats/Common/interface/ContainerMaskTraits.h:43:32: warning: deleted member function should be public [modernize-use-equals-delete]
    const ContainerMaskTraits& operator=(const ContainerMaskTraits&) = delete;  // stop default
```

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: DataFormats/Common/interface/ContainerMaskTraits.h [hunk 1]
```diff
@@ -35,13 +35,14 @@ namespace edm {
       return iElement - &(iContainer->front());
     }
 
-  private:
-    //virtual ~ContainerMaskTraits();
+  public:
```
Comment (makortel) [inline DataFormats/Common/interface/ContainerMaskTraits.h:38]: No need to repeat
```suggestion
```

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34668	ee54ef7
- technical

**Comment by makortel** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

Technical PR. Failing in the PR test is not related to this PR.

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
merge

**Pull request merged**